File: easy/intersection.pl
use strict;
open( INFILE, "<", $ARGV[0] )
  or die("Cannot open file $ARGV[0] for reading: $!");
while ( my $line = <INFILE> ) {
    chomp($line);
    my ( $s, $t ) = split( /;/, $line );
    my @v = split( /,/, $t );
    my @r;
    foreach my $i ( split( /,/, $s ) ) {
        shift @v while ( @v && @v[0] < $i );
        if ( @v && @v[0] == $i ) {
            push @r, shift @v;
        }
    }
    printf( "%s\n", join( ',', @r ) );
}
close(INFILE);

File: easy/intersection.rb
File.open(ARGV[0]).each_line do |line|
  s = line.chomp.split(';')
  a = s[0].split(',').map(&:to_i)
  b = s[1].split(',').map(&:to_i)
  j = 0
  r = []
  a.each do |x|
    j += 1 while j < b.length && x > b[j]
    if j < b.length && x == b[j]
      r << x.to_s
      j += 1
    end
  end
  puts r.join(',')
end
